server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -222,15 +223,18 @@ public interface IndexCommitListener {
          * {@link IndexCommitRef} files to be deleted from disk until the reference is closed. As such, the listener must close the
          * reference as soon as it is done with it.
          *
+         * @param shardId the {@link ShardId} of shard
+         * @param primaryTerm a {@link LongSupplier} for the shard's primary term
          * @param indexCommitRef a reference on the newly created index commit
          */
-        void onNewCommit(ShardId shardId, Engine.IndexCommitRef indexCommitRef);
+        void onNewCommit(ShardId shardId, LongSupplier primaryTerm, Engine.IndexCommitRef indexCommitRef);
```

Comment:
Does this need to be a `LongSupplier` here? Would it not suffice to just pass the current primary term?

Reply:
No real need for a supplier so I pushed 37ac3675f1108bb7e336eafc4baa671a0322ae24